app/controllers/recipes_controller.rb

```diff
@@ -1,6 +1,13 @@
 class RecipesController < ApplicationController
+
   def index
-  end
+    if ingredient_id = params["ingredient"]
+      ingredient_id.to_i
```

Comment:
It's not necessary and to be honest do nothing. To make it works you should do:

ingredient_id = ingredient_id.to_i